From ecb7a2bbcf68921cdca7aed2740561b6bb14970b Mon Sep 17 00:00:00 2001
From: Peter Robinson <pbrobinson@gmail.com>
Date: Fri, 6 Mar 2015 10:31:19 +0000
Subject: [PATCH 04/14] Switch am335x_evm.h to use config_distro_defaults and
 config_distro_bootcmd. Add scriptaddr to DEFAULT_LINUX_BOOT_ENV, as
 scriptaddr is used for boot scripts in config_distro_defaults.

---
 include/configs/am335x_evm.h      | 26 +++++++++++++++++---------
 include/configs/ti_armv7_common.h |  1 +
 2 files changed, 18 insertions(+), 9 deletions(-)

diff --git a/include/configs/am335x_evm.h b/include/configs/am335x_evm.h
index a87059c..3fb175e 100644
--- a/include/configs/am335x_evm.h
+++ b/include/configs/am335x_evm.h
@@ -18,6 +18,21 @@
 
 #include <configs/ti_am335x_common.h>
 
+#include <config_distro_defaults.h>
+
+#define BOOTENV_BOOT_PARTITIONS "1 2"
+
+#define BOOTENV_INIT_COMMAND "run findfdt ;"
+#ifdef CONFIG_NAND
+#define BOOTENV_POST_COMMAND "run nandboot ;"
+#endif
+
+#define BOOT_TARGET_DEVICES(func) \
+	func(MMC, mmc, 0) \
+	func(MMC, mmc, 1)
+
+#include <config_distro_bootcmd.h>
+
 #ifndef CONFIG_SPL_BUILD
 #ifndef CONFIG_FIT
 # define CONFIG_FIT
@@ -181,17 +196,10 @@
 		"if test $fdtfile = undefined; then " \
 			"echo WARNING: Could not determine device tree to use; fi; \0" \
 	NANDARGS \
-	DFUARGS
+	DFUARGS \
+	BOOTENV
 #endif
 
-#define CONFIG_BOOTCOMMAND \
-	"run findfdt; " \
-	"run mmcboot;" \
-	"setenv mmcdev 1; " \
-	"setenv bootpart 1:2; " \
-	"run mmcboot;" \
-	"run nandboot;"
-
 /* NS16550 Configuration */
 #define CONFIG_SYS_NS16550_COM1		0x44e09000	/* Base EVM has UART0 */
 #define CONFIG_SYS_NS16550_COM2		0x48022000	/* UART1 */
diff --git a/include/configs/ti_armv7_common.h b/include/configs/ti_armv7_common.h
index 110a4f8..b6e37b3 100644
--- a/include/configs/ti_armv7_common.h
+++ b/include/configs/ti_armv7_common.h
@@ -54,6 +54,7 @@
 #define DEFAULT_LINUX_BOOT_ENV \
 	"loadaddr=0x82000000\0" \
 	"kernel_addr_r=0x82000000\0" \
+	"scriptaddr=0x82000000\0" \
 	"fdtaddr=0x88000000\0" \
 	"fdt_addr_r=0x88000000\0" \
 	"rdaddr=0x88080000\0" \
-- 
2.3.2

